addon/road_base.py

```diff
@@ -336,39 +357,39 @@ def compare_boundaries_with_s(self, s, s_boundaries_next):
 
         return smaller, idx_sorted[0]
 
-    def get_road_vertices_edges_faces(self, strips, road_sample_points):
+    def get_road_vertices_edges_faces(self, road_sample_points):
         '''
-            Return blender compatible vertices, edges and faces concatenated and
-            calculated from road sample points.
+           generate mesh from samplepoints
         '''
         vertices = []
         edges = []
         faces = []
         idx_vertex = 0
-        for idx_strip, strip in enumerate(strips):
-            if not (strip.type == 'line' and strip.type_road_mark == 'none'):
-                for idx_face_strip in range(len(road_sample_points[2*idx_strip])):
-                    samples_right = road_sample_points[2*idx_strip+1][idx_face_strip]
-                    samples_left = road_sample_points[2*idx_strip][idx_face_strip]
-                    num_vertices = len(samples_left) + len(samples_right)
-                    vertices += samples_right + samples_left[::-1]
-                    edges += [[idx_vertex+n, idx_vertex+n+1] for n in range(num_vertices-1)] \
-                            + [[idx_vertex+num_vertices-1, idx_vertex]]
-                    faces += [[idx_vertex+n for n in range(num_vertices)]]
-
-                    idx_vertex += num_vertices
-
+        point_index = 0
+        while point_index < len(road_sample_points):
+            for idx_face_strip in range(len(road_sample_points[point_index])):
+                if not road_sample_points[point_index][0]:
+                    continue
+                samples_right = road_sample_points[point_index + 1][idx_face_strip]
+                samples_left = road_sample_points[point_index][idx_face_strip]
+                num_vertices = len(samples_left) + len(samples_right)
+                vertices += samples_right + samples_left[::-1]
+                edges += [[idx_vertex + n, idx_vertex + n + 1] for n in range(num_vertices - 1)] \
+                         + [[idx_vertex + num_vertices - 1, idx_vertex]]
+                faces += [[idx_vertex + n for n in range(num_vertices)]]
+                idx_vertex += num_vertices
+            point_index = point_index + 2
         return vertices, edges, faces
 
     def get_face_materials(self, strips, strips_s_boundaries):
         '''
             Return dictionary with index of faces for each material.
         '''
-        materials = {'asphalt': [], 'road_mark': [], 'grass': []}
+        materials = {'asphalt': [], 'road_mark': [], 'grass': [], 'double_yellow_line': []}
```

Comment:
I guess here we need
{'asphalt': [], 'road_mark_white': [], 'road_mark_yellow': [], 'grass': []}